# <center> Lab 2. Analysis of data on passengers of the liner "Titanic" </center>

**The task requires to use Pandas to answer several questions according to the Kaggle repository (https://www.kaggle.com/c/titanic/data) (you do not need to download the data - they are already in the directory).**


In [ ]:
import numpy as np
import pandas as pd
%matplotlib inline

**Read data from a file into memory as a Pandas.DataFrame object**

**Data Access on google drive**, if you open the notebook in **google colab**, and not on PC, you can get by mounting google drive

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive') 

In [ ]:
!ls gdrive/MyDrive/TEACHING/IntroDataScience/intro_to_data_science/Lab_topic_1/data

In [1]:
# the path to the data folder on my google drive (edit according for your case) 
data_folder = "gdrive/MyDrive/TEACHING/IntroDataScience/intro_to_data_science/Lab_topic_1/data" 

In [ ]:
#data = pd.read_csv('data/titanic_train.csv', index_col='PassengerId')
data = pd.read_csv(data_folder+'/titanic_train.csv', index_col='PassengerId')

**Data are presented in the form of a table. Let's look at the first 5 lines:**

In [ ]:
data.head(5)

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

In [ ]:
data.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

**For example, select passengers who boarded in Cherbourg (Embarked = C) and paid more than $ 200 for a ticket (fare> 200).**

Make sure you understand how this design works. <br>
If not, see how the expression in square brackets is calculated.

In [ ]:
data[(data['Embarked'] == 'C') & (data.Fare > 200)].head()

Survived  Pclass  \
PassengerId                     
119                 0       1   
259                 1       1   
300                 1       1   
312                 1       1   
378                 0       1   

                                                        Name     Sex   Age  \
PassengerId                                                                  
119                                 Baxter, Mr. Quigg Edmond    male  24.0   
259                                         Ward, Miss. Anna  female  35.0   
300          Baxter, Mrs. James (Helene DeLaudeniere Chaput)  female  50.0   
312                               Ryerson, Miss. Emily Borie  female  18.0   
378                                Widener, Mr. Harry Elkins    male  27.0   

             SibSp  Parch    Ticket      Fare            Cabin Embarked  
PassengerId                                                              
119              0      1  PC 17558  247.5208          B58 B60        C  
259              0      0  PC 17755  512.3292              NaN        C  
300              0      1  PC 17558  247.5208          B58 B60        C  
312              2      2  PC 17608  262.3750  B57 B59 B63 B66        C  
378              0      2    113503  211.5000              C82        C

**You can sort these people by reducing the ticket price.**

In [ ]:
data[(data['Embarked'] == 'C') & 
     (data['Fare'] > 200)].sort_values(by='Fare',
                               ascending=False).head()

Survived  Pclass                                   Name     Sex  \
PassengerId                                                                    
259                 1       1                       Ward, Miss. Anna  female   
680                 1       1     Cardeza, Mr. Thomas Drake Martinez    male   
738                 1       1                 Lesurer, Mr. Gustave J    male   
312                 1       1             Ryerson, Miss. Emily Borie  female   
743                 1       1  Ryerson, Miss. Susan Parker "Suzette"  female   

              Age  SibSp  Parch    Ticket      Fare            Cabin Embarked  
PassengerId                                                                    
259          35.0      0      0  PC 17755  512.3292              NaN        C  
680          36.0      0      1  PC 17755  512.3292      B51 B53 B55        C  
738          35.0      0      0  PC 17755  512.3292             B101        C  
312          18.0      2      2  PC 17608  262.3750  B57 B59 B63 B66        C  
743          21.0      2      2  PC 17608  262.3750  B57 B59 B63 B66        C

**Example of creating a feature.**

In [ ]:
def age_category(age):
    '''
    < 30 -> 1
    >= 30, <55 -> 2
    >= 55 -> 3
    '''
    if age < 30:
        return 1
    elif age < 55:
        return 2
    else:
        return 3

In [ ]:
age_categories = [age_category(age) for age in data.Age]

In [ ]:
data['Age_category'] = age_categories

**Another way is through `apply`.**

In [ ]:
data['Age_category'] = data['Age'].apply(age_category)

**1. How many men / women were on board?**
- 412 men and 479 women
- 314 men and 577 women
- 479 men and 412 women
- 577 men and 314 women

In [ ]:
# your code is here

**2. Print the distribution of the variable `Pclass` (socio-economic status) and the same distribution, only for men / women separately. How many men were in the 2nd grade?**
- 104
- 108
- 112
- 125

In [ ]:
# your code is here

**3. What are the median and standard deviation values (`Fare`)? Round to 2 decimal places.**
- Median - 14.45, standard deviation - 49.69
- Median - 15.1, standard deviation - 12.15
- Median - 13.15, standard deviation - 35.3
- Median - 17.43, standard deviation - 39.1

In [ ]:
# your code is here

**4. Is it true that people under 30 survived more often than people over 60? What is the proportion of survivors in both groups?**
- 22.7% among the young and 40.6% among the old
- 40.6% among the young and 22.7% among the old
- 35.3% among the young and 27.4% among the old
- 27.4% among the young and 35.3% among the old

In [ ]:
# your code is here

**5. Is it true that women survived more often than men? What is the proportion of survivors in both groups?** 
- 30.2% among men and 46.2% among women 
- 35.7% among men and 74.2% among women
- 21.1% among men and 46.2% among women
- 18.9% among men and 74.2% among women

In [ ]:
# your code is here

**6. Find the most popular names among male Titanic passengers**
- Charles
- Thomas
- William
- John

In [ ]:
# your code is here

**7. Compare graphically the distribution of ticket prices and age of the rescued and the dead. The average age of the dead is higher, right?**
- Yes
- No

In [ ]:
# your code is here

**8. How does the average age of men / women differ depending on the class of service? Choose the correct statements:**
- On average, men of the 1st class are older than 40 years
- On average, 1st grade women are over 40 years old
- Men of all classes are on average older than women of the same class
- On average, people in the 1st grade are older than in the 2nd, as well as older representatives of the 3rd grade

In [ ]:
# your code is here